libyul/optimiser/FunctionSpecializer.cpp

```diff
@@ -0,0 +1,149 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+
+#include <libyul/optimiser/FunctionSpecializer.h>
+
+#include <libyul/optimiser/ASTCopier.h>
+#include <libyul/optimiser/NameCollector.h>
+#include <libyul/optimiser/NameDispenser.h>
+
+#include <libyul/AST.h>
+#include <libyul/YulString.h>
+#include <libsolutil/CommonData.h>
+
+#include <boost/algorithm/cxx11/any_of.hpp>
```

Comment:
Can this be replaced by any of the new range stuff?

Reply:
We can replace `boost::range::any_of` with `ranges::any_of`, but it feels weird using the same function from range-v3 instead of boost.

Reply:
But I'll replace it here.

Reply:
I think the advantage is that once we have c++20 it's kind of "done". And this way we can keep reducing our boost footprint :p